server/src/main/java/org/opensearch/action/search/TransportDeletePitAction.java

```diff
@@ -0,0 +1,134 @@
+    private static final Logger logger = LogManager.getLogger(TransportDeletePitAction.class);
+
+    @Inject
+    public TransportDeletePitAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NamedWriteableRegistry namedWriteableRegistry,
+        TransportSearchAction transportSearchAction,
+        ClusterService clusterService,
+        SearchTransportService searchTransportService
+    ) {
+        super(DeletePitAction.NAME, transportService, actionFilters, DeletePitRequest::new);
+        this.namedWriteableRegistry = namedWriteableRegistry;
+        this.transportSearchAction = transportSearchAction;
+        this.clusterService = clusterService;
+        this.searchTransportService = searchTransportService;
+    }
+
+    /**
+     * Invoke 'delete all pits' or 'delete list of pits' workflow based on request
+     */
+    @Override
+    protected void doExecute(Task task, DeletePitRequest request, ActionListener<DeletePitResponse> listener) {
+        List<String> pitIds = request.getPitIds();
+        if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+            deleteAllPits(listener);
+        } else {
+            deletePits(listener, request);
+        }
+    }
+
+    /**
+     * Deletes list of pits, return success if all reader contexts are deleted ( or not found ).
+     */
+    private void deletePits(ActionListener<DeletePitResponse> listener, DeletePitRequest request) {
+        Map<String, List<SearchContextIdForNode>> nodeToContextsMap = new HashMap<>();
+        for (String pitId : request.getPitIds()) {
+            SearchContextId contextId = SearchContextId.decode(namedWriteableRegistry, pitId);
+            for (SearchContextIdForNode contextIdForNode : contextId.shards().values()) {
+                List<SearchContextIdForNode> contexts = nodeToContextsMap.getOrDefault(contextIdForNode.getNode(), new ArrayList<>());
+                contexts.add(contextIdForNode);
+                nodeToContextsMap.put(contextIdForNode.getNode(), contexts);
+            }
+        }
+        ActionListener<Integer> deleteListener = ActionListener.wrap(r -> {
+            if (r == nodeToContextsMap.size()) {
+                listener.onResponse(new DeletePitResponse(true));
+            } else {
+                logger.debug(
+                    () -> new ParameterizedMessage(
+                        "Delete PITs failed. Cleared contexts in {} nodes out of {}",
+                        r,
+                        nodeToContextsMap.size()
+                    )
+                );
+                listener.onResponse(new DeletePitResponse(false));
+            }
+        }, e -> {
+            logger.error("Delete PITs failed ", e);
+            listener.onResponse(new DeletePitResponse(false));
+        });
+        SearchUtils.deletePitContexts(nodeToContextsMap, deleteListener, clusterService.state(), searchTransportService);
+    }
```

Comment:
Can we have cases where some PITs on a particular node is missing or already deleted?

Reply:
Yeah we've added tests as well , where in such cases we mark the operation as succeeded